projects/packages/jetpack-mu-wpcom/src/features/custom-css/custom-css.php

```diff
@@ -0,0 +1,1247 @@
+<?php // phpcs:ignore WordPress.Files.FileName.InvalidClassFileName
+
+// phpcs:disable Universal.Files.SeparateFunctionsFromOO.Mixed -- TODO: Move classes to appropriately-named class files.
+// @phan-file-suppress PhanRedefinedClassReference,PhanRedefineFunction,PhanRedefineClass
+/**
+ * Alternate Custom CSS source for 4.7 compat.
+ *
+ * @since 4.4.2
+ *
+ * @package automattic/jetpack
+ */
+
+use Automattic\Jetpack\Assets;
+
+if ( ! class_exists( 'Jetpack_Custom_CSS_Enhancements' ) ) {
+	/**
+	 * Class Jetpack_Custom_CSS_Enhancements
+	 */
+	class Jetpack_Custom_CSS_Enhancements {
+
+		/**
+		 * Set up the actions and filters needed for our compatability layer on top of core's Custom CSS implementation.
+		 */
+		public static function add_hooks() {
+			add_action( 'init', array( __CLASS__, 'init' ) );
+			add_action( 'customize_controls_enqueue_scripts', array( __CLASS__, 'customize_controls_enqueue_scripts' ) );
+			add_action( 'customize_register', array( __CLASS__, 'customize_register' ) );
+			add_filter( 'map_meta_cap', array( __CLASS__, 'map_meta_cap' ), 20, 2 );
+			add_action( 'customize_preview_init', array( __CLASS__, 'customize_preview_init' ) );
+			add_filter( '_wp_post_revision_fields', array( __CLASS__, 'wp_post_revision_fields' ), 10, 2 );
+			add_action( 'load-revision.php', array( __CLASS__, 'load_revision_php' ) );
+
+			add_action( 'wp_enqueue_scripts', array( __CLASS__, 'wp_enqueue_scripts' ) );
+
+			// Handle Sass/LESS.
+			add_filter( 'customize_value_custom_css', array( __CLASS__, 'customize_value_custom_css' ), 10, 2 );
+			add_filter( 'customize_update_custom_css_post_content_args', array( __CLASS__, 'customize_update_custom_css_post_content_args' ), 10, 3 );
+			add_filter( 'update_custom_css_data', array( __CLASS__, 'update_custom_css_data' ), 10, 2 );
+
+			// Stuff for stripping out the theme's default stylesheet...
+			add_filter( 'stylesheet_uri', array( __CLASS__, 'style_filter' ) );
+			add_filter( 'safecss_skip_stylesheet', array( __CLASS__, 'preview_skip_stylesheet' ) );
+
+			// Stuff for overriding content width...
+			add_action( 'customize_preview_init', array( __CLASS__, 'preview_content_width' ) );
+			add_filter( 'jetpack_content_width', array( __CLASS__, 'jetpack_content_width' ) );
+			add_filter( 'editor_max_image_size', array( __CLASS__, 'editor_max_image_size' ), 10, 3 );
+			add_action( 'template_redirect', array( __CLASS__, 'set_content_width' ) );
+			add_action( 'admin_init', array( __CLASS__, 'set_content_width' ) );
+			add_action( 'jetpack_modules_loaded', array( __CLASS__, 'custom_css_loaded' ) );
+		}
+
+		/**
+		 * Things that we do on init.
+		 */
+		public static function init() {
```

Comment:
It might be worth rewriting much of the init function, in terms of registering and enqueuing scripts, so that it uses the Assets package? This was suggested for Calypsoify - https://github.com/Automattic/jetpack/pull/37339#discussion_r1598584806 - here is how it looks now: https://github.com/Automattic/jetpack/blob/6b12d65e303d82b2443200dbfc212ab2c091be44/projects/packages/calypsoify/src/class-jetpack-calypsoify.php#L73

The issues with the JETPACK__VERSION constant could be removed that way too, or defining a package version constant in this class then referencing that with `self::PACKAGE_VERSION`.